### ISSUE TITLE
Compatibility with GLPI 9.1

### ISSUE THREAD

**Issue by Sismic** [ddurieux/glpi_monitoring#240]
we can't update GLPI 0.90 to 9.1. Because of an incompatibility with the plugin, migration is impossible and it is a shame.

**Comment by keyser75000** [comment]
It can work in 9.1.2 with little modifications in setup.php for both plugins monitoring and webservices.
In file setup.php, just modify test in fonction "plugin_<PLUGIN>_check_prerequisites()" :
` if (version_compare(GLPI_VERSION,'0.85','lt') || version_compare(GLPI_VERSION,'9.2','ge')) {`

**Comment by voileux** [comment]
could you please upgrade your plugin to be compatible with GLPI 9.1.

**Comment by ddurieux** [comment]
Yes, will do it quickly

**Comment by voileux** [comment]
Could you provide an update on this issue ?

**Comment by ddurieux** [comment]
I will do this this week end ;)

**Comment by voileux** [comment]
I'm looking forward to Monday  ;) 
Have a good week-end

**Comment by keyser75000** [comment]
Hi,

I'm testing GLPI 9.2 and I still need monitoring.
I managed to make monitoring work. Some method are removed since 9.1 others are deprecated (still work but will be removed).

I found only these methods that break compatibilty :
- in componentscatalog.class.php:191, edit static hosts in components catalog.
```
//      Dropdown::showAllItems('items_id');
      Dropdown::showSelectItemFromItemTypes(['items_id']);
```


- in componentscatalog.class.php:1047, synthesis in components catalog.
For "Start Date field" :
```
//Html::showDateFormItem("date_start", date('Y-m-d H:i:s', date('U') - (24 * 3600 * 7)));
Html::showDateTimeField("date_start",
                              array('value'      => date('Y-m-d H:i:s', date('U') - (24 * 3600 * 7)),
                                    'timestep'   => 1,
                                    'maybeempty' => true
));
```
and for "End date field" :
```
//Html::showDateFormItem("date_end", date('Y-m-d'));
Html::showDateTimeField("date_end",
                              array('value'      => date('Y-m-d H:i:s'),
                                    'timestep'   => 1,
                                    'maybeempty' => true
));
```

Others methods are deprecated ("openArrowMassives()" and "closeArrowMassives") but still work. I didn't find others but I didn't check everything.
For my use (managed Shinken config, computers state, resources state, alerts) it still works perfectly.
It allows me to wait for an future update (and probably migrate to Alignak ?).

**Comment by FrichardF** [comment]
Bonjour,

Est ce qu'il y a du nouveau vis à vis du Plugin Monitoring ?

Je dois rendre un rapport lundi et cela ne marche toujours pas :+1: 
![image](https://user-images.githubusercontent.com/33471508/32527533-c17cb6b6-c42e-11e7-9d84-dba28dccbe4b.png)

Une piste ? Un nouveau plugin ?

Merci d'avance !

**Comment by FrichardF** [comment]
Hello and sorry to rewrite this post (better in english isn't it ?

On GLPI 9.1.6, I try to install : 
     Injection de fichiers | 2.4.2 | GPLv2+ | Activé | W. Nouh, R. Collet, N. Mahu-Lasson, X. Caillaud
     Monitoring | 0.90+1.0 | AGPLv3+ | Activé | D. DURIEUX
     OCS Inventory NG | 1.3.3 | GPLv2+ | Activé | G. Dubois, R. Collet, N. Mahu-Lasson, D. Durieux, X. Caillaud, W. Nouh, A. Jaouen
     Services Web | 1.7. | GPLv2+ | Activé | Remi Collet, Nelly Mahu-Lasson

I experience some problemes withthe Monitoring Problem which are not well displayed on GLPI.
(Please have a look on the screenshot below)

Is there some update of this plugin, or something to do for make that working ?

Is there someone working on that ?

Thanks by advance

Richard

![image](https://user-images.githubusercontent.com/33471508/32537196-05b853c0-c462-11e7-8efc-6819382990a8.png)

**Comment by ddurieux** [comment]
Check the profiles

**Comment by FrichardF** [comment]
Could you explain me how to do as I am a beginner ?
Perhaps you have an exemple ?

Thanks by advance
Richard

**Comment by keyser75000** [comment]
Check in "Administration > Profils > YOUR_PROFILE".
![image](https://user-images.githubusercontent.com/19589135/32538229-977fd3d4-c465-11e7-922a-ade5740ff611.png)

**Comment by ddurieux** [comment]
Ok, so that's not normal :(

**Comment by keyser75000** [comment]
@ddurieux it's an example for @FrichardF  :)

**Comment by ddurieux** [comment]
Ah ok sorry :p

**Comment by FrichardF** [comment]
Hello,

I have the same than you : Another Idea ?

Is there an update ?

Thanks by advance
![image](https://user-images.githubusercontent.com/33471508/32538946-2e087494-c468-11e7-965c-c495a23208f1.png)

**Comment by keyser75000** [comment]
Is there any error in debug mode ?
Are you in 9.2 version or 9.1.x ?

**Comment by FrichardF** [comment]
Debug mode ? Sorry, as I am a real beginner.  I am on 9.1.6.
Thanks for your time.

**Comment by keyser75000** [comment]
Do you have Shinken installed ? Does it communicate with GLPI, through webservices plugin (arbiter log) ?

**Comment by FrichardF** [comment]
Many thanks for your help and supports !!!

**Comment by FrichardF** [comment]
Hello,

After so many days in front of my computers (and 2 complete nights) :

So, I have solved the pending problem on shinken : just let localhost in all the .cfg files from manuals :
![image](https://user-images.githubusercontent.com/33471508/33008534-b966c260-cdd3-11e7-9688-c3b7db442ff1.png)


There is a problem left, an idea to solve that ? 
![image](https://user-images.githubusercontent.com/33471508/33008394-380b26de-cdd3-11e7-8adc-ec3ce1143fca.png)

Many thanks for your help and supports !!!

**Comment by Megachip** [comment]
@keyser75000 could you provide a patch which @ddurieux could include? Would be nice to see monitoring compatible with 9.1 and 9.2

**Comment by keyser75000** [comment]
@FrichardF : I think you're Shinken <=> GLPI communication is not correctly installed :
- In arbiter log I don't see any communication between import-glpi module and webservices GLPI plugin.
- Also in arbiter log, I don't see ws-arbiter (is 7760 port listening ?)
- You're broker module glpidb doesn't seem to work : "No module named MySQLdb" (python module for mysql).
- php library "php_xmlrpc.dll" doesn't seem to load.
You're Shinken is working in standalone mode (localhost is configured by default).

In summary, you have to check you're whole installation.
Have a look to this doc : [http://sysadmin.cool/2016/03/installation-glpi-monitoring](url)
You should have a look to glpi forum too. Post preferably on glpi forum (and/or Shinken one). This issue was for glpi compatibility with 9.1).

**Comment by keyser75000** [comment]
@Megaship I made a request for 9.1.
As I said, I managed to make it work with 9.2, by replacing some olds methods. But I don't use at this time 9.2 in production, so I'm not sure there's not side effects.

You can also use my solutions in #226.
